core/controllers/base.py

```diff
@@ -295,8 +299,81 @@ def dispatch(self):
                 self.handle_exception(e, self.app.debug)
                 return
 
+        schema_validation_succeeded = True
+        try:
+            self.validate_args_schema()
+        except self.InvalidInputException as e:
+            self.handle_exception(e, self.app.debug)
+            schema_validation_succeeded = False
+        # TODO(#13155): Remove this clause once all the handlers have had
+        # schema validation implemented.
+        except NotImplementedError as e:
+            self.handle_exception(e, self.app.debug)
+            schema_validation_succeeded = False
+
+        if schema_validation_succeeded is False:
+            return
+
         super(BaseHandler, self).dispatch()
 
+    def validate_args_schema(self):
+        """Validates schema for controller layer handler class arguments.
+
+        Raises:
+            InvalidInputException. Schema validation failed.
+            NotImplementedError. Schema is not provided in handler class.
+        """
+        handler_class_name = self.__class__.__name__
+        request_method = self.request.environ['REQUEST_METHOD']
+        handler_args = self.payload
+        url_path_args = self.request.route_kwargs
+        handler_class_names_with_no_schema = (
+            payload_validator.HANDLER_CLASS_NAMES_WITH_NO_SCHEMA)
+
+        if handler_class_name in handler_class_names_with_no_schema:
+            return
+
+        if self.URL_PATH_ARGS_SCHEMAS is None:
+            raise NotImplementedError(
+                'Missing schema for url path args in %s '
+                'handler class' % (handler_class_name))
+
+        # For html handlers, extra args are allowed (to accommodate
+        # e.g. utm parameters which are not used by the backend but
+        # needed for analytics).
+        extra_args_are_allowed = False
+        if (self.GET_HANDLER_ERROR_RETURN_TYPE == 'html' and
+                request_method == 'GET'):
+            extra_args_are_allowed = True
+
+        # Validation for url path elements.
+        schema_for_url_path_args = self.URL_PATH_ARGS_SCHEMAS
+        errors = payload_validator.validate(
+            url_path_args, schema_for_url_path_args, extra_args_are_allowed)
+        if errors:
+            raise self.InvalidInputException('\n'.join(errors))
+
+        # Validation for body params and url query string params.
+        if request_method in ['GET', 'DELETE']:
+            handler_args = {}
+            for arg in self.request.arguments():
+                handler_args[arg] = self.request.get(arg)
+        if handler_args is None:
+            return
+
+        try:
+            schema_for_handler_args = self.HANDLER_ARGS_SCHEMAS
+            schema_for_request_method = schema_for_handler_args[request_method]
```

Comment:
Is it resolved?

Reply:
I have looked into the "files" section in github and there was no open comments as because they are now outdated.
Since they are present here (main tab) so I missed to mark it as done. (sorry for tihs.)

Reply:
Done